serializers/src/main/java/io/pravega/schemaregistry/codec/Codecs.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.codec;
+
+import com.fasterxml.jackson.databind.util.ByteBufferBackedInputStream;
+import io.pravega.schemaregistry.contract.data.CodecType;
+import lombok.Getter;
+import org.apache.commons.io.IOUtils;
+import org.xerial.snappy.Snappy;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Utility class for creating codecs for none, snappy or gzip. 
+ */
+public enum Codecs {
+    None(Constants.NOOP),
+    GzipCompressor(Constants.GZIP_CODEC), 
+    SnappyCompressor(Constants.SNAPPY_CODEC);
+
+    @Getter
+    private final Codec codec;
+    
+    Codecs(Codec codec) {
+        this.codec = codec;  
+    }
+
+    private static class Noop implements Codec {
+        private static final CodecType CODEC_TYPE_NONE = new CodecType(Constants.NONE);
+
+        @Override
+        public String getName() {
+            return CODEC_TYPE_NONE.getName();
+        }
+
+        @Override
+        public CodecType getCodecType() {
+            return CODEC_TYPE_NONE;
+        }
+
+        @Override
+        public void encode(ByteBuffer data, ByteArrayOutputStream bos) {
```

Comment:
Noop does nothing, not even write to bos. 
In the AbstractSerializer we explicitly pass the same bos to underlying serializer implementation which writes the serialized bytes to the bos and encoder does nothing at all.

Reply:
Ok. That's not what I would guess for a codec. The others are gzip and snappy compress. Surely there should be a plain passthrough for just encoding the data according to the schema right? If noop does nothing should we rename it or make it private to avoid this confusion? Like maybe "nodata" or something?

Reply:
i originally had a method interface that looked as follows
`ByteBuffer encode(ByteBuffer buffer)`
This allowed the no op to return original buffer. 

we changed this to `void encode(ByteBuffer data, ByteArrayOutputStream bos)`.
this change to the signature was done to reduce one extra copy of data that we were making in the actual encoding case. 

basically, now we pass an outputstream where we may have already written the encoding header so that we dont copy over encoded bytes from bytebuffer into a buffer with encoding header.

Reply:
I have made the change to copy the buffer into the outputstream as part of noop codec.
however, serializer doesnt call the encode method on noop codec.